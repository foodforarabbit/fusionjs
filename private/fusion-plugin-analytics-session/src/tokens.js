// @flow
import {createToken} from 'fusion-core';

import type {Token} from 'fusion-core';
import type {CookieModuleType, CookieTypeType, PluginType} from './types';

export const AnalyticsCookieTypeToken: Token<
  Array<CookieTypeType> | CookieTypeType
> = createToken('AnalyticsSessionCookieType');
export const AnalyticsCookieModuleToken: Token<CookieModuleType> = createToken(
  'AnalyticsCookieModule'
);
export const AnalyticsSessionToken: Token<PluginType> = createToken(
  'AnalyticsSession'
);
