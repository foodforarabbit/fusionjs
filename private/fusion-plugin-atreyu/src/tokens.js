// @flow
import {createToken} from 'fusion-core';
import Atreyu from '@uber/atreyu';

export const AtreyuToken = createToken<typeof Atreyu>('AtreyuToken');
export const AtreyuConfigToken = createToken<Object>('AtreyuConfigToken');
export const AtreyuOptionsToken = createToken<Object>('AtreyuOptionsToken');
export const AtreyuClientToken = createToken<typeof Atreyu>(
  'AtreyuClientToken'
);
