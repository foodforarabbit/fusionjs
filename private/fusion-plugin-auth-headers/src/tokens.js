// @flow

import type {Token} from 'fusion-core';
import {createToken} from 'fusion-core';

import type {AuthHeadersService} from './server';

export const AuthHeadersToken: Token<AuthHeadersService> = createToken(
  'AuthHeadersToken'
);

export const AuthHeadersUUIDConfigToken: Token<string> = createToken(
  'AuthHeadersUUIDConfigToken'
);
export const AuthHeadersEmailConfigToken: Token<string> = createToken(
  'AuthHeadersUUIDConfigToken'
);
export const AuthHeadersTokenConfigToken: Token<string> = createToken(
  'AuthHeadersTokenConfigToken'
);
export const AuthHeadersRolesConfigToken: Token<string> = createToken(
  'AuthHeadersRolesConfigToken'
);
export const AuthHeadersGroupsConfigToken: Token<string> = createToken(
  'AuthHeadersGroupsConfig'
);
