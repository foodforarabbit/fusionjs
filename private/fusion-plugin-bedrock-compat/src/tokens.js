// @flow
/* eslint-env browser */
import {createToken} from 'fusion-core';

import type {
  InitializeServerType,
  BedrockCompatPluginServiceType,
} from './types';

export const InitializeServerToken = createToken<InitializeServerType>(
  'InitializeServer'
);
export const BedrockCompatToken = createToken<BedrockCompatPluginServiceType>(
  'BedrockCompat'
);
