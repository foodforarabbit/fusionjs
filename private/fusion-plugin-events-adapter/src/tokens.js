/* eslint-env browser */
import {createToken} from 'fusion-tokens';

const EventsAdapterToken = createToken('EventAdapterToken');
const EventsAdapterAnalyticsToken = createToken('EventsAdapterAnalyticsToken');

export {EventsAdapterToken, EventsAdapterAnalyticsToken};
