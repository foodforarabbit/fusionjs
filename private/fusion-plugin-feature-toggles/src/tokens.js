// @flow

import {createToken} from 'fusion-core';
import type {Token} from 'fusion-core';

import type {
  FeatureToggleConfigType,
  IFeatureTogglesClient,
  FeatureTogglesServiceType,
  ClientConfigType,
} from './types.js';

export const FeatureTogglesClientToken: Token<
  Class<IFeatureTogglesClient>
> = createToken('FeatureTogglesClientToken');

export const FeatureTogglesClientConfigToken: Token<ClientConfigType> = createToken(
  'FeatureTogglesClientConfigToken'
);

export const FeatureTogglesTogglesConfigToken: Token<
  Array<FeatureToggleConfigType | string>
> = createToken('FeatureTogglesTogglesConfigToken');

export const FeatureTogglesToken: Token<FeatureTogglesServiceType> = createToken(
  'FeatureTogglesToken'
);
