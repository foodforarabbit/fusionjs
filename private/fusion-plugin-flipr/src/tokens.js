// @flow
import {createToken} from 'fusion-core';

export const FliprToken = createToken('FliprToken');

export const FliprClientToken = createToken('FliprClientToken');
export const FliprConfigToken = createToken('FliprConfigToken');
