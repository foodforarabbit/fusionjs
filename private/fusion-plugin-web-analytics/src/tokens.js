// @flow
import {createToken} from 'fusion-core';
import {FliprToken} from '@uber/fusion-plugin-flipr';

import type {Token} from 'fusion-core';
import type {WebAnalyticsPluginServiceType} from './types';

export const UberWebAnalyticsToken: Token<WebAnalyticsPluginServiceType> = createToken(
  'UberWebAnalyticsToken'
);
// TODO: support analyticsState
export const UberWebAnalyticsStateToken: Token<any> = createToken(
  'UberWebAnalyticsStateToken'
);
export const UberWebAnalyticsFliprToken: typeof FliprToken = createToken(
  'UberWebAnalyticsFliprToken'
);
export const UberWebAnalyticsFliprConfigToken: Token<Object> = createToken(
  'UberWebAnalyticsFliprConfigToken'
);
