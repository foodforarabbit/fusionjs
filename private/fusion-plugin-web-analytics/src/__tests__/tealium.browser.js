// @flow
/* eslint-env browser */
/* global global */
import App from 'fusion-core';
import {getSimulator} from 'fusion-test-utils';
import UniversalEvents, {
  UniversalEventsToken,
} from 'fusion-plugin-universal-events';

import Plugin from '../index.js';
import {UberWebAnalyticsToken} from '../tokens';

import {
  testTealiumConfig,
  testTealiumReduxStore,
} from '../fixtures/test-tealium-config';

function appendConfig(config) {
  const mockConfigEl = document.createElement('div');
  mockConfigEl.id = '__ANALYTICS_CONFIG__';
  mockConfigEl.textContent = JSON.stringify(config);
  const HeadEl: any = document.head;
  HeadEl.appendChild(mockConfigEl);
  return {cleanupEl: () => HeadEl.removeChild(mockConfigEl)};
}

test('Tealium', async () => {
  const {cleanupEl} = appendConfig(testTealiumConfig);
  global.postMessage = jest.fn();
  global.utag = {link: jest.fn()};

  expect.assertions(3);

  const app = new App('TestEl', el => el);
  app.register(UniversalEventsToken, UniversalEvents);
  app.register(UberWebAnalyticsToken, Plugin);
  app.middleware(
    {webAnalytics: UberWebAnalyticsToken},
    ({webAnalytics}) => async (ctx, next) => {
      webAnalytics.eventContext.setReduxState(testTealiumReduxStore);
      webAnalytics.track('GET_OFFERS_SUCCESS', {
        type: 'GET_OFFERS_SUCCESS',
        payload: {},
      });
      return next();
    }
  );
  const sim = getSimulator(app);

  await sim.render('/test');

  expect(global.postMessage.mock.calls[0][0]).toMatchSnapshot();
  expect(global.postMessage.mock.calls[1][0]).toMatchSnapshot();
  expect(global.utag.link.mock.calls[0][0]).toMatchSnapshot();

  cleanupEl();
});
