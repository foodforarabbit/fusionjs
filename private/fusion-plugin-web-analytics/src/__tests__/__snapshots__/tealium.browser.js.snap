// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tealium 1`] = `
Object {
  "_eventContext": Object {
    "payload": Object {
      "payload": Object {},
      "type": "GET_OFFERS_SUCCESS",
    },
    "redux": Object {
      "offers": Object {
        "data": Array [
          Object {
            "description": "foobar barfoo",
          },
        ],
      },
    },
    "window": Object {
      "document": Object {
        "cookie": Array [],
      },
      "referrer": undefined,
    },
  },
  "eventKey": "GET_OFFERS_SUCCESS",
  "eventPayload": Object {
    "payload": Object {},
    "type": "GET_OFFERS_SUCCESS",
  },
  "type": "fusion-plugin-web-analytics.dispatch",
}
`;

exports[`Tealium 2`] = `
Object {
  "destinationKey": "tealium",
  "dispatchValue": Object {
    "firstOfferFetched": "foobar barfoo",
    "foo": Object {
      "a": 1,
      "b": 3,
      "c": Object {
        "c1": "m",
        "c2": "p",
        "c3": "q",
      },
      "d": true,
    },
  },
  "eventKey": "GET_OFFERS_SUCCESS",
  "type": "fusion-plugin-web-analytics.destinationDispatch",
}
`;

exports[`Tealium 3`] = `
Object {
  "firstOfferFetched": "foobar barfoo",
  "foo": Object {
    "a": "1",
    "b": "3",
    "c": Object {
      "c1": "m",
      "c2": "p",
      "c3": "q",
    },
    "d": "true",
  },
}
`;
